core/ordersync/ordersync.go

```diff
@@ -458,6 +468,41 @@ func parseResponseWithSubprotocol(subprotocol Subprotocol, providerID peer.ID, r
 	}, nil
 }
 
+type FirstRequestsForSubprotocols struct {
+	MetadataForSubprotocol []json.RawMessage `json:"metadata"`
+}
+
+// createFirstRequestForAllSubprotocols creates an initial ordersync request that
+// contains metadata for all of the ordersync subprotocols. This allows more guarantees
+// to be made about the types of orders that will be received from a given ordersync
+// subprotocol, and these guarantees allow ordersync connections to be pruned much
+// more aggressively.
```

Comment:
```suggestion
// createFirstRequestForAllSubprotocols creates an initial ordersync request that
// contains metadata for all of the ordersync subprotocols.
```

nit: I know what you're talking about because I have additional context from our conversations in Slack, but I don't think you've provided enough context in the comment itself. It's not clear how creating an initial request allows us to prune connections more aggressively, and that pruning is not implemented anywhere yet. Are you okay with removing this part of the comment for now? When we do implement more aggressive connection pruning, I think this comment (or something like it) belongs at the point in the code where pruning occurs.